### PR TITLE
 Support multi-processing on python3.8

### PR DESCRIPTION
**Pull request by janjaapdriessen**

### CODE REVIEW

**Comment by janjaapdriessen** [comment]
@tisto I closed [the previous PR](https://github.com/plone/plone.recipe.codeanalysis/pull/229) to please the "Plone Contributors Agreement verifier". I down-pinned flake8 to a version that doesn't pull in the `zipp` dependency. Travis is now green on python2.7 and python3.6+.

File: plone/recipe/codeanalysis/__init__.py [hunk 1]
```diff
@@ -293,6 +293,13 @@ def parse_command_line_arguments(options):
     return options
 
 
+def _taskrunner(klass, options, lock, status):
+    check = klass(options, lock)
+    if check.enabled:
+        if not check.run():
+            status.value = False
+
+
```
Comment (gyst) [inline plone/recipe/codeanalysis/__init__.py:302]: I don't mind inner methods; but if you do I'm surprised the method gets factored out but the inner class does not.

**Review by gyst — APPROVED** [review]
LGTM but I did not actually test this.

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 language: python
 python:
   - 2.7
-  - 3.4
-  - 3.5
   - 3.6
+  - 3.7
```
Comment (tisto) [inline .travis.yml:5]: We might want to reflect the support for 3.7/3.8 in our setup.py.

Comment (janjaapdriessen) [reply .travis.yml:5]: Done in 77eba7b6a80ced0a277d604542e5612835337b11

**Comment by tisto** [comment]
@janjaapdriessen thanks for your contribution!

When trying to do a release I did a buildout run and ran into a problem with rubygems:

```
Installing rubygems.
Downloading https://rubygems.org/rubygems/rubygems-3.1.4.zip
ERROR:  While executing gem ... (Errno::EACCES)
    Permission denied @ apply2files - /Library/Ruby/Gems/2.6.0/specifications/default/bundler-1.17.2.gemspec
rubygems:
rubygems: Command failed with exit code 1: [u'ruby', u'setup.rb', u'all', u'--prefix=/Users/timo/workspace/plone/plone.recipe.codeanalysis/parts/rubygems', u'--no-rdoc', u'--no-ri']
While:
  Installing rubygems.
Error: System error
make: *** [.installed.cfg] Error 1
```

I really dislike installing tools via buildout. Maybe it is time to remove those checks from p.r.codeanalysis. Those kind of checks belongs to the frontend part. Everything we do in Plone with JS is horribly outdated...

**Comment by janjaapdriessen** [comment]
@tisto I feel your pain. Shall I make a PR to eradicate scsslint and all it's ruby dependencies altogether?

**Comment by tisto** [comment]
@janjaapdriessen I'd say so. p.r.codeanlysis should restrict itself to Python checks. The longer we try to maintain those deps, the more pain we will see. Mixing Python and JS was never a good idea (my fault) and we should stop putting more effort into a fight that we are going to lose anyway.

@gyst would you be ok with that move?

**Comment by gyst** [comment]
Note that the rubygems problem is a regression caused by changing this line:
```buildout -N -t 3 -c $CONF || buildout -N -t 3 -c $CONF```
into a single buildout invocation. That hack was there for specifically this reason. I did not comment on its removal since Travis appeared to be ok with it.

I don't see the rationale for checking "only python". That would imply removing most of the checkers, including zpt, chameleon, i18n and all the javascript checkers. One might as well ditch plone.recipe.codeanalysis and use black or whatever directly.

**Comment by tisto** [comment]
@gyst with "python only" I meant everything except JS checkers, which, as said, I believe belong to the frontend stack (whatever that might be).

**Comment by janjaapdriessen** [comment]
Can we continue this discussion in https://github.com/plone/plone.recipe.codeanalysis/pull/231 please?

**Pull request merged**